src/Components/Testimonials/TestimonialsForm.js

```diff
@@ -1,32 +1,96 @@
-import React, { useState } from 'react';
-import '../FormStyles.css';
+import {useEffect, useState} from 'react';
+import {Formik, Form, Field} from 'formik';
+import { ErrorsForm } from '../common/messagesForm';
+import { CKEditor } from '@ckeditor/ckeditor5-react';
+import { UrlInput } from '../common/getUrlInputFile';
+import ClassicEditor from '@ckeditor/ckeditor5-build-classic';
+import { validateFormTestimonials, replaceCKeditor } from '../common/validateFormTestimonials';
+import {getTestimonial,createTestimonial,modifiedTestimonial} from '../../Services/testimonialService';
+const TestimonialForm = ({testimonialId}) => {
 
-const TestimonialForm = () => {
-    const [initialValues, setInitialValues] = useState({
-       name: '',
-       description: '' 
-    });
+  const [imgState,setImgState] = useState();
+  const [dataForm, setDataForm] = useState({
+    name:'',
+    description:'',
+    image: ''
+  });
 
-    const handleChange = (e) => {
-        if(e.target.name === 'name'){
-            setInitialValues({...initialValues, name: e.target.value})
-        } if(e.target.name === 'description'){
-            setInitialValues({...initialValues, description: e.target.value})
-        }
-    }
+  useEffect(() => {
+    if(testimonialId) setDataForm(getTestimonial(testimonialId));
+  }, [testimonialId]);
 
-    const handleSubmit = (e) => {
-        e.preventDefault();
-        console.log(initialValues);
-    }
+  const handleChaneCkEditor = (editor, setFieldValue) => {
+    setFieldValue("description", editor.getData())
+  };
 
-    return (
-        <form className="form-container" onSubmit={handleSubmit}>
-            <input className="input-field" type="text" name="name" value={initialValues.name} onChange={handleChange} placeholder="Testimonial Title"></input>
-            <input className="input-field" type="text" name="description" value={initialValues.description} onChange={handleChange} placeholder="Testimonial description"></input>
+  const handleChangeinputFile = (e, setFieldValue) =>{ 
+    UrlInput(e, setImgState,setFieldValue);
+  };
+
+  const placeholder = "Write some testimonial description";
+  return (
+    <div className='form-container'>
+      <Formik
+        initialValues={{
+          ...dataForm
+        }}
+        validate={(values)=>validateFormTestimonials(values)}
+        onSubmit={(values)=>{
+          let valuesForm=replaceCKeditor(values)
```

Comment:
this is updatedValues instead valuesForm because both values and valuesForm are form values, the diference is that one is as it is and the second one is updated or cleaned value. So the name should be updatedValues or cleanedValues